files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md

```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
 ### 参数
 
 - `searchElement`
-
-  - : 需要查找的元素值。
-
-    > **备注：** 使用 `includes()` 比较字符串和字符时是区分大小写的。
-
+  - : 需要查找的值。
 - `fromIndex` {{optional_inline}}
-  - : 从`fromIndex` 索引处开始查找 `searchElement`。如果为负值，则按升序从 `array.length + fromIndex` 的索引开始搜（即使从末尾开始往前跳 `fromIndex` 的绝对值个索引，然后往后搜寻）。默认为 0。
+  - ：从零开始搜索的索引，[转换为整数](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number#转换为整数)。
+    - 负索引从数组末尾开始计数——假如 `fromIndex < 0`，`fromIndex + array.length` 被调用。然而，在这种情况下，仍然从前到后搜索数组。
+    - 如果 `fromIndex < -array.length` 或者省略 `fromIndex`，则使用 `0`，导致搜索整个数组。
+    - 如果 `fromIndex >= array.length`，则不会搜索数组并返回 `false`。
 
 ### 返回值
 
-返回一个布尔值 {{jsxref("Boolean")}} 。
-如果在数组中（或 `fromIndex` 指定的范围中）找到了 `searchElement`，则返回 `true`，否则返回 `false`。
+一个布尔值，如果在数组中找到 `searchElement` 值，则该值为 `true`（或者在指定的索引 `fromIndex` 所指示的数组部分中找到 `searchElement` 值）。
+
+## 描述
 
-0 的值将全部视为相等，与符号无关（即 -0 与 0 和 +0 相等），但 `false` 不被认为与 0 相等。
+`includes()` 方法使用[零值相等](/zh-CN/docs/Web/JavaScript/Equality_comparisons_and_sameness#零值相等)算法将 `searchElement` 与数组中的元素进行比较。0 值都被认为是相等的，不管符号是什么。（即 `-0` 等于 `0`），但 `false` 不被认为与 `0` 相同。[`NaN`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/NaN) 可以被正确搜索。
 
-> **备注：** 技术上来讲，`includes()` 使用 [`零值相等`](/zh-CN/docs/Web/JavaScript/Equality_comparisons_and_sameness#%E9%9B%B6%E5%80%BC%E7%9B%B8%E7%AD%89) 算法来确定是否找到给定的元素。
+当使用[稀疏数组](/zh-cn/docs/Web/JavaScript/Guide/Indexed_collections#稀疏数组)时， `includes()` 方法迭代空槽，就像它们的值 `undefined` 一样。
+
+`includes()` 方法是[通用的](/zh-cn/docs/Web/JavaScript/Reference/Global_Objects/Array#通用的数组方法)。它只期望 `this` 值具有 `length` 属性和整型键属性。
```

Comment:
```suggestion
`includes()` 方法是[通用的](/zh-cn/docs/Web/JavaScript/Reference/Global_Objects/Array#通用的数组方法)。它只期望 `this` 值具有 `length` 属性和整数键属性。
```

Heads up: this note is present on almost every array method page. Translators will want to make sure that they use the same message format for every page.